src/routes/Home.jsx

```diff
@@ -1,38 +1,72 @@
-// React Imports
-import React from 'react';
-// Material UI Imports
-import Box from '@mui/material/Box';
-import Container from '@mui/material/Container';
-import Paper from '@mui/material/Paper';
-
-/**
- * Home Page - Page that generates the home page of the PASS website/app
- *
- * @memberof Pages
- * @name Home
- */
-
-const Home = () => (
-  /* eslint-disable jsx-a11y/label-has-associated-control */
-  <Container component="main" maxWidth="s">
-    <Box
-      sx={{
-        marginTop: 18,
-        display: 'flex',
-        flexDirection: 'column',
-        alignItems: 'center',
-        textAlign: 'center',
-        padding: '20px'
-      }}
-    >
-      <Paper elevation={2} sx={{ display: 'inline-block', mx: '2px', padding: '20px' }}>
-        <section id="home">
-          <h1>HOME PAGE</h1>
-        </section>
-      </Paper>
-    </Box>
-  </Container>
-  /* eslint-enable jsx-a11y/label-has-associated-control */
-);
+import React, { useState, useEffect, useMemo, useContext } from 'react';
+import { useSession } from '@inrupt/solid-ui-react';
+
+import {
+  InboxMessageContext,
+  SelectUserContext,
+  UserListContextProvider,
+  SignedInUserContext
+} from '../contexts';
+
+// Utility Imports
+import {
+  createDocumentContainer,
+  createPublicContainer,
+  createOutbox,
+  createInbox,
+  getInboxMessageTTL
+} from '../utils';
+import { updateUserActivity } from '../model-helpers';
+import Layout from '../layouts/Layouts';
+import AppRoutes from '../AppRoutes';
+
+const Home = () => {
+  const { session } = useSession();
+  const { podUrl } = useContext(SignedInUserContext);
+
+  const [selectedUser, setSelectedUser] = useState('');
+  const [loadMessages, setLoadMessages] = useState(true);
+  const [inboxList, setInboxList] = useState([]);
+
+  const selectedUserObject = useMemo(() => ({ selectedUser, setSelectedUser }), [selectedUser]);
+  const inboxMessageObject = useMemo(() => ({ inboxList, setInboxList }), [inboxList]);
+
+  /**
+   * A function that generates a Users container if logging in for the first
+   * time and initalizes the list of users from Solid
+   *
+   * @function fetchData
+   */
+  const fetchData = async () => {
+    await Promise.all([
+      updateUserActivity(session, podUrl),
+      createPublicContainer(session, podUrl),
+      createDocumentContainer(session, podUrl),
+      createOutbox(session, podUrl),
+      createInbox(session, podUrl),
+      getInboxMessageTTL(session, inboxList).then((messages) => {
```

Comment:
getInboxMessageTTL will be updated with getMessageTTL from PR #191, might need to deal with merge conflict once it's merged.